<!doctype html>
<html lang=en>
<head>
<meta charset=utf-8>
<title>E. Coli Whole Cell Model by Covert Lab - Ciro Santilli</title>
<meta property="og:title" content="E. Coli Whole Cell Model by Covert Lab - Ciro Santilli">
<meta property="og:type" content="website">
<meta property="og:image" content="https://upload.wikimedia.org/wikipedia/commons/9/94/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_mass_fraction_summary.svg">
<meta property="og:url" content="https://cirosantilli.com/e-coli-whole-cell-model-by-covert-lab">
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.10.1/css/all.min.css" integrity="sha512-9my9Mb2+0YO+I4PUCSwUYO7sEK21Y0STBAiFEYoWtd2VzLEZZ4QARDrZ30hdM1GlioHJ8o8cWQiy8IAb1hy/Hg==" crossorigin="anonymous" referrerpolicy="no-referrer">
<link rel="canonical" href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab">
<style>@import "_obb/dist/ourbigbook.css";

</style>
<link rel="stylesheet" type="text/css" href="_raw/main.css">
<link rel="shortcut icon" type="image/x-icon" href="https://raw.githubusercontent.com/cirosantilli/media/master/ID_photo_of_Ciro_Santilli_taken_in_2013_square_398.jpg">
</head>
<body>
<header>
<div class="brand-group">
<a href="." class="brand"><img src="https://raw.githubusercontent.com/cirosantilli/media/master/ID_photo_of_Ciro_Santilli_taken_in_2013_right_eye_200_100.jpg" loading="lazy" alt="ID photo of Ciro Santilli taken in 2013 right eye">Ciro Santilli</a>
<a href="https://ourbigbook.com/cirosantilli"><img src="https://raw.githubusercontent.com/cirosantilli/media/master/ourbigbook-logo-v1.svg" loading="lazy" alt="OurBigBook logo">OurBigBook.com</a>
<a class="font-awesome-container" href="https://stackoverflow.com/users/895245"><i class="fab fa-stack-overflow fa-fw icon"></i></a>
<a class="font-awesome-container" href="https://github.com/cirosantilli"><i class="fab fa-github fa-fw icon"></i></a>
<a class="font-awesome-container" href="https://www.linkedin.com/in/cirosantilli"><i class="fab fa-linkedin fa-fw icon"></i></a>
<a class="font-awesome-container" href="https://www.youtube.com/c/CiroSantilli"><i class="fab fa-youtube fa-fw icon"></i></a>
<a class="font-awesome-container" href="https://twitter.com/cirosantilli"><i class="fab fa-twitter fa-fw icon"></i></a>
<a class="font-awesome-container" href="https://www.zhihu.com/people/cirosantilli/activities"><i class="fab fa-zhihu fa-fw icon"></i></a>
<a class="font-awesome-container" href="https://www.weibo.com/p/1005055601627311"><i class="fab fa-weibo fa-fw icon"></i></a>
<a href="sponsor"><span class="icon">$£</span>&nbsp;Sponsor</a>
<a href="https://github.com/cirosantilli/china-dictatorship"><span class="icon">中国</span>独裁统治&nbsp;China Dictatorship 新疆改造中心、六四事件、法轮功、郝海东、709大抓捕、2015巴拿马文件 邓家贵、低端人口、西藏骚乱</a>
</div>
</header>
<main class="ourbigbook">
<div class="h top" id="e-coli-whole-cell-model-by-covert-lab"><div class="notnav"><h1><a href="e-coli-whole-cell-model-by-covert-lab-split">E. Coli Whole Cell Model by Covert Lab <span class="meta">(CovertLab/WholeCellEcoliRelease)</span></a></h1></div><nav class="h-nav h-nav-toplevel"><div class="nav ancestors"><a  href="#_ancestors">&nbsp;...</a><a href="taxonomy"> Taxonomy</a><a href="taxonomy#species"> Species</a><a href="taxonomy#bacteria"> Bacteria</a><a href="taxonomy#list-of-bacteria"> List of bacteria</a><a href="taxonomy#escherichia-coli"> Escherichia coli</a><a href="taxonomy#e-coli-whole-cell-simulation"> E. Coli whole cell simulation</a></div><div class="nav"><a href="#_toc" class="toc"></a><a href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab"><img src="_obb/logo.svg" class="logo" alt=""> OurBigBook.com</a><span class="tags"> Tags: <a href="articles">The best articles by Ciro Santilli</a></span><a class="split" href="e-coli-whole-cell-model-by-covert-lab-split"></a><span class="metrics"><span class="wcntr"> Words: 4k</span><span class="dcnt"> Articles: 11</span></span></div></nav></div><div class="p" id="_3"><a href="https://github.com/CovertLab/WholeCellEcoliRelease">github.com/CovertLab/WholeCellEcoliRelease</a> is a <a href="cell#whole-cell-simulation">whole cell simulation</a> model created by <a href="university#covert-lab">Covert Lab</a> and other collaborators.</div><div class="p" id="_4">The project is written in <a href="programming-language#python-programming-language">Python</a>, hurray!</div><div class="p" id="_5">But according to te <a href="software#readme">README</a>, it seems to be the use a <a href="software#code-drop">code drop</a> model with on-request access to master. <a href="ciro-santilli">Ciro Santilli</a> asked at <a href="https://github.com/CovertLab/WholeCellEcoliRelease/discussions/23">rationale on GitHub discussion</a>, and they confirmed as expected that it is to:<div class="list"><ul id="_6"><li id="_7">to prevent their <a href="education#academic-publishing">publication</a> ideas from being stolen. Who would steal publication ideas with public proof in an issue tracker without crediting original authors? <a href="education#academia-is-broken">Academia is broken</a>. Academia should be the most open form of knowledge sharing. But instead we get this silly competition for publication points.</li><li id="_8">to prevent noise from non-collaborators. But they only get like 2 issues as year on such a meganiche subject... Did you know that you can ignore people, and even block them if they are particularly annoying? Much more likely is that no one will every hear about your project and that it will die with its last graduate student slave.</li></ul></div></div><div class="p" id="_9">The project is a followup to the earlier <a href="taxonomy#m-genitalium-whole-cell-model-by-covert-lab">M. genitalium whole cell model by Covert lab</a> which modelled <a href="taxonomy#mycoplasma-genitalium">Mycoplasma genitalium</a>. <a href="taxonomy#escherichia-coli">E. Coli</a> has 8x more genes (500 vs 4k), but it the undisputed <a href="taxonomy#bacteria">bacterial</a> <a href="biology#model-organism">model organism</a> and as such has been studied much more thoroughly. It also reproduces faster than Mycoplasma (20 minutes vs a few hours), which is a huge advantages for validation/exploratory <a href="science#experiment">experiments</a>.</div><div class="p" id="_10">The project has a partial dependency on the <a href="law#proprietary-software">proprietary</a> <a href="computer-science#optimization-software">optimization software</a> <a href="computer-science#cplex">CPLEX</a> which is <a href="law#freeware">freeware</a>, for students, not sure what it is used for exactly, from the comment in the <code>requirements.txt</code> the dependency is only partial.</div><div class="p" id="_11">This project makes <a href="ciro-santilli">Ciro Santilli</a> think of the <a href="taxonomy#escherichia-coli">E. Coli</a> as an <a href="computer-science#optimization-problem">optimization problem</a>. Given such external nutrient/temperature condition, which <a href="dna">DNA</a> sequence makes the cell grow the fastest? Balancing <a href="molecular-biology#metabolite">metabolites</a> feels like designing a <a href="video-game#factorio">Factorio</a> speedrun.</div><div class="p" id="_12">There is one major thing missing thing in the current model: <a href="dna#promoter-genetics">promoters</a>/<a href="dna#transcription-factor">transcription factor</a> interactions are not modelled due to lack/low quality of experimental data: <a href="https://github.com/CovertLab/WholeCellEcoliRelease/issues/21">github.com/CovertLab/WholeCellEcoliRelease/issues/21</a>. They just have a magic direct "<a href="dna#transcription-factor">transcription factor</a> to <a href="dna#gene">gene</a>" relationship, encoded at <a href="https://github.com/CovertLab/WholeCellEcoliRelease/blob/7e4cc9e57de76752df0f4e32eca95fb653ea64e4/reconstruction/ecoli/flat/foldChanges.tsv">reconstruction/ecoli/flat/foldChanges.tsv</a> in terms of type "if this is present, such protein is expressed 10x more". <a href="dna#transcription-unit">Transcription units</a> are not implemented at all it appears.</div><div class="p" id="_13">Everything in this section refers to version <a href="https://github.com/CovertLab/WholeCellEcoliRelease/tree/7e4cc9e57de76752df0f4e32eca95fb653ea64e4">7e4cc9e57de76752df0f4e32eca95fb653ea64e4</a>, the code drop from November 2020, and was tested on <a href="systems-programming#ubuntu">Ubuntu</a> 21.04 with a docker install of <code>docker.pkg.github.com/covertlab/wholecellecolirelease/wcm-full</code> with image id 502c3e604265, unless otherwise noted.</div><div class="toc-container" id="_toc"><ul><li class="has-child toplevel"><div class="title-div"><div class="arrow"><div></div></div><span class="not-arrow"><a class="title toc" href="#_toc"> Table of contents</a><input class="search" placeholder="🔍 Search. Shortcut: / (slash)"><span class="hover-metadata"><span class="metrics"><span class="wcntr"> 4k</span><span class="dcnt"> 11</span></span></span></span></div><ul><li><div id="_toc/install-and-first-run"><div class="arrow"><div></div></div><span class="not-arrow"><a href="#install-and-first-run"><i class="n">1. </i>Install and first run</a><span class="hover-metadata"><a href="#_toc" class="u"> E. Coli Whole Cell Model by Covert Lab</a><span class="metrics"><span class="wcntr"> 327</span></span></span></span></div></li><li class="has-child"><div id="_toc/output-overview"><div class="arrow"><div></div></div><span class="not-arrow"><a href="#output-overview"><i class="n">2. </i>Output overview</a><span class="hover-metadata"><a href="#_toc" class="u"> E. Coli Whole Cell Model by Covert Lab</a><span class="metrics"><span class="wcntr"> 681</span><span class="dcnt"> 1</span></span></span></span></div><ul><li><div id="_toc/mass-fraction-summary-plot-analysis"><div class="arrow"><div></div></div><span class="not-arrow"><a href="#mass-fraction-summary-plot-analysis"><i class="n">2.1. </i>Mass fraction summary plot analysis</a><span class="hover-metadata"><a href="#_toc/output-overview" class="u"> Output overview</a><span class="metrics"><span class="wcntr"> 351</span></span></span></span></div></li></ul><li class="has-child"><div id="_toc/run-variants"><div class="arrow"><div></div></div><span class="not-arrow"><a href="#run-variants"><i class="n">3. </i>Run variants</a><span class="hover-metadata"><a href="#_toc" class="u"> E. Coli Whole Cell Model by Covert Lab</a><span class="metrics"><span class="wcntr"> 928</span><span class="dcnt"> 2</span></span></span></span></div><ul><li><div id="_toc/default-run-variant"><div class="arrow"><div></div></div><span class="not-arrow"><a href="#default-run-variant"><i class="n">3.1. </i>Default run variant</a><span class="hover-metadata"><a href="#_toc/run-variants" class="u"> Run variants</a><span class="metrics"><span class="wcntr"> 80</span></span></span></span></div></li><li><div id="_toc/time-series-run-variant"><div class="arrow"><div></div></div><span class="not-arrow"><a href="#time-series-run-variant"><i class="n">3.2. </i>Time series run variant</a><span class="hover-metadata"><a href="#_toc/run-variants" class="u"> Run variants</a><span class="metrics"><span class="wcntr"> 741</span></span></span></span></div></li></ul><li><div id="_toc/other-run-variants"><div class="arrow"><div></div></div><span class="not-arrow"><a href="#other-run-variants"><i class="n">4. </i>Other run variants</a><span class="hover-metadata"><a href="#_toc" class="u"> E. Coli Whole Cell Model by Covert Lab</a><span class="metrics"><span class="wcntr"> 75</span></span></span></span></div></li><li class="has-child"><div id="_toc/source-code-overview"><div class="arrow"><div></div></div><span class="not-arrow"><a href="#source-code-overview"><i class="n">5. </i>Source code overview</a><span class="hover-metadata"><a href="#_toc" class="u"> E. Coli Whole Cell Model by Covert Lab</a><span class="metrics"><span class="wcntr"> 1k</span><span class="dcnt"> 1</span></span></span></span></div><ul><li><div id="_toc/condition"><div class="arrow"><div></div></div><span class="not-arrow"><a href="#condition"><i class="n">5.1. </i>Condition</a><span class="hover-metadata"><a href="#_toc/source-code-overview" class="u"> Source code overview</a><span class="metrics"><span class="wcntr"> 471</span></span></span></span></div></li></ul><li><div id="_toc/model-validation"><div class="arrow"><div></div></div><span class="not-arrow"><a href="#model-validation"><i class="n">6. </i>Model validation</a><span class="hover-metadata"><a href="#_toc" class="u"> E. Coli Whole Cell Model by Covert Lab</a><span class="metrics"><span class="wcntr"> 5</span></span></span></span></div></li><li><div id="_toc/publications"><div class="arrow"><div></div></div><span class="not-arrow"><a href="#publications"><i class="n">7. </i>Publications</a><span class="hover-metadata"><a href="#_toc" class="u"> E. Coli Whole Cell Model by Covert Lab</a><span class="metrics"><span class="wcntr"> 78</span></span></span></span></div></li></ul></li></ul></div><div class="h" id="install-and-first-run"><div class="notnav"><h2><a href="e-coli-whole-cell-model-by-covert-lab/install-and-first-run"><span class="number">1. </span>Install and first run</a></h2><span class="hover-meta"> <a class="split" href="e-coli-whole-cell-model-by-covert-lab/install-and-first-run"></a><span class="metrics"><span class="wcntr"> Words: 327</span></span></span></div><nav class="h-nav h-nav-toplevel"><div class="nav"><a href="" class="u"> E. Coli Whole Cell Model by Covert Lab</a><a href="#_toc/install-and-first-run" class="toc"></a><a href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab/install-and-first-run"><img src="_obb/logo.svg" class="logo" alt=""> OurBigBook.com</a></div></nav></div><div class="p" id="_14">At <a href="https://github.com/CovertLab/WholeCellEcoliRelease/tree/7e4cc9e57de76752df0f4e32eca95fb653ea64e4">7e4cc9e57de76752df0f4e32eca95fb653ea64e4</a> you basically need to use the <a href="systems-programming#docker-software">Docker</a> image on <a href="systems-programming#ubuntu">Ubuntu</a> 21.04 due to <a href="programming-language#pip-package-manager">pip</a> breaking changes... (not their fault). Perhaps <a href="programming-language#pyenv">pyenv</a> would solve things, but who has the patience for that?!?!</div><div class="p" id="_15">The Docker setup from README does just work. The image download is a bit tedius, as it requires you to create a GitHub API key as described in the README, but there must be reasons for that.</div><div class="p" id="_16">Once the image is downloaded, you really want to run is from the root of the source tree:<div class="code" id="_17"><div><pre><code>sudo docker run --name=wcm -it -v "$(pwd):/wcEcoli" docker.pkg.github.com/covertlab/wholecellecolirelease/wcm-full</code></pre></div></div>This mounts the host source under <code>/wcEcoli</code>, so you can easily edit and view output images from your host. Once inside Docker we can compile, run the simulation, and analyze results with:<div class="code" id="_18"><div><pre><code>make clean compile &amp;&amp;
python runscripts/manual/runFitter.py &amp;&amp;
python runscripts/manual/runSim.py &amp;&amp;
python runscripts/manual/analysisVariant.py &amp;&amp;
python runscripts/manual/analysisCohort.py &amp;&amp;
python runscripts/manual/analysisMultigen.py &amp;&amp;
python runscripts/manual/analysisSingle.py</code></pre></div></div>The meaning of each of the analysis commands is described at <a href="#output-overview">Section 2. "Output overview"</a>.</div><div class="p" id="_19">As a <a href="systems-programming#docker-software">Docker</a> refresher, after you stop the container, e.g. by restarting your computer or running <code>sudo docker stop wcm</code>, you can get back into it with:<div class="code" id="_20"><div><pre><code>sudo docker start wcm
sudo docker run -it wcm bash</code></pre></div></div></div><div class="p" id="_21"><code>runscripts/manual/runFitter.py</code> takes about 15 minutes, and it generates files such as <code>reconstruction/ecoli/dataclasses/process/two_component_system.py</code> (<a href="https://github.com/CovertLab/WholeCellEcoliRelease/issues/20">related</a>) which is required to run the simulation, it is basically a part of the build.</div><div class="p" id="_22"><code>runSim.py</code> does the main simulation, progress output contains lines of type:<div class="code" id="_23"><div><pre><code>Time (s)  Dry mass     Dry mass      Protein          RNA    Small mol     Expected
              (fg)  fold change  fold change  fold change  fold change  fold change
========  ========  ===========  ===========  ===========  ===========  ===========
    0.00    403.09        1.000        1.000        1.000        1.000        1.000
    0.20    403.18        1.000        1.000        1.000        1.000        1.000</code></pre></div></div>and then it ended on the <a href="ciro-santilli-s-hardware#lenovo-thinkpad-p51-2017">Lenovo ThinkPad P51 (2017)</a> at:<div class="code" id="_24"><div><pre><code> 2569.18    783.09        1.943        1.910        2.005        1.950        1.963

Simulation finished:
 - Length: 0:42:49
 - Runtime: 0:09:13</code></pre></div></div>when the cell had almost doubled, and presumably divided in 42 minutes of simulated time, which could make sense compared to the 20 under optimal conditions.</div><div class="h" id="output-overview"><div class="notnav"><h2><a href="e-coli-whole-cell-model-by-covert-lab/output-overview"><span class="number">2. </span>Output overview</a></h2><span class="hover-meta"> <a class="split" href="e-coli-whole-cell-model-by-covert-lab/output-overview"></a><span class="metrics"><span class="wcntr"> Words: 681</span><span class="dcnt"> Articles: 1</span></span></span></div><nav class="h-nav h-nav-toplevel"><div class="nav"><a href="" class="u"> E. Coli Whole Cell Model by Covert Lab</a><a href="#_toc/output-overview" class="toc"></a><a href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab/output-overview"><img src="_obb/logo.svg" class="logo" alt=""> OurBigBook.com</a></div></nav></div><div class="p" id="_25">Run output is placed under <code>out/</code>:</div><div class="p" id="_26">Some of the output data is stored as <code>.cpickle</code> files. To observe those files, you need the original Python classes, and therefore you have to be inside Docker, from the host it won't work.</div><div class="p" id="_27">We can list all the plots that have been produced under <code>out/</code> with<div class="code" id="_28"><div><pre><code>find -name '*.png'</code></pre></div></div>Plots are also available in <a href="computer#scalable-vector-graphics">SVG</a> and <a href="computer#pdf">PDF</a> formats, e.g.:<div class="list"><ul id="_29"><li id="_30"><a href="computer#portable-network-graphics">PNG</a>: <code>./out/manual/plotOut/low_res_plots/massFractionSummary.png</code></li><li id="_31"><a href="computer#scalable-vector-graphics">SVG</a>: <code>./out/manual/plotOut/svg_plots/massFractionSummary.svg</code> The SVGs write text as polygons, see also: <a href="computer#svg-fonts">SVG fonts</a>.</li><li id="_32"><a href="computer#pdf">PDF</a>: <code>./out/manual/plotOut/massFractionSummary.pdf</code></li></ul></div></div><div class="p" id="_33">The output directory has a hierarchical structure of type:<div class="code" id="_34"><div><pre><code>./out/manual/wildtype_000000/000000/generation_000000/000000/</code></pre></div></div>where:<div class="list"><ul id="_35"><li id="_36"><code>wildtype_000000</code>: variant conditions. <code>wildtype</code> is a human readable label, and <code>000000</code> is an index amongst the possible <code>wildtype</code> conditions. For example, we can have different simulations with different nutrients, or different <a href="dna">DNA</a> sequences. An example of this is shown at <a href="#run-variants">run variants</a>.</li><li id="_37"><code>000000</code>: initial random seed for the initial cell, likely fed to <a href="programming-language#numpy">NumPy</a>'s <code>np.random.seed</code></li><li id="_38"><code>genereation_000000</code>: this will increase with generations if we simulate multiple cells, which is supported by the model</li><li id="_39"><code>000000</code>: this will presumably contain the cell index within a generation</li></ul></div></div><div class="p" id="_40">We also understand that some of the top level directories contain summaries over all cells, e.g. the <code>massFractionSummary.pdf</code> plot exists at several levels of the hierarchy:<div class="code" id="_41"><div><pre><code>./out/manual/plotOut/massFractionSummary.pdf
./out/manual/wildtype_000000/plotOut/massFractionSummary.pdf
./out/manual/wildtype_000000/000000/plotOut/massFractionSummary.pdf
./out/manual/wildtype_000000/000000/generation_000000/000000/plotOut/massFractionSummary.pdf</code></pre></div></div></div><div class="p" id="_42">Each of thoes four levels of <code>plotOut</code> is generated by a different one of the analysis scripts:<div class="list"><ul id="_43"><li id="_44"><code>./out/manual/plotOut</code>: generated by <code>python runscripts/manual/analysisVariant.py</code>. Contains comparisons of different variant conditions. We confirm this by looking at the results of <a href="#run-variants">run variants</a>.</li><li id="_45"><code>./out/manual/wildtype_000000/plotOut</code>: generated by <code>python runscripts/manual/analysisCohort.py --variant_index 0</code>. TODO not sure how to differentiate between two different labels e.g. <code>wildtype_000000</code> and <code>somethingElse_000000</code>. If <code>-v</code> is not given, a it just picks the first one alphabetically. TODO not sure how to automatically generate all of those plots without inspecting the directories.</li><li id="_46"><code>./out/manual/wildtype_000000/000000/plotOut</code>: generated by <code>python runscripts/manual/analysisMultigen.py --variant_index 0 --seed 0</code></li><li id="_47"><code>./out/manual/wildtype_000000/000000/generation_000000/000000/plotOut</code>: generated by <code>python runscripts/manual/analysisSingle.py --variant_index 0 --seed 0 --generation 0 --daughter 0</code>. Contains information about a single specific cell.</li></ul></div></div><div class="h" id="mass-fraction-summary-plot-analysis"><div class="notnav"><h3><a href="e-coli-whole-cell-model-by-covert-lab/mass-fraction-summary-plot-analysis"><span class="number">2.1. </span>Mass fraction summary plot analysis</a></h3><span class="hover-meta"> <a class="split" href="e-coli-whole-cell-model-by-covert-lab/mass-fraction-summary-plot-analysis"></a><span class="metrics"><span class="wcntr"> Words: 351</span></span></span></div><nav class="h-nav h-nav-toplevel"><div class="nav"><a href="#output-overview" class="u"> Output overview</a><a href="#_toc/mass-fraction-summary-plot-analysis" class="toc"></a><a href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab/mass-fraction-summary-plot-analysis"><img src="_obb/logo.svg" class="logo" alt=""> OurBigBook.com</a></div></nav></div><div class="p" id="_48">Let's look into a sample plot, <code>out/manual/plotOut/svg_plots/massFractionSummary.svg</code>, and try to understand as much as we can about what it means and how it was generated.</div><div class="p" id="_49">This plot contains how much of each type of mass is present in all cells. Since we simulated just one cell, it will be the same as the results for that cell.</div><div class="p" id="_50">We can see that all of them grow more or less <a href="linear-algebra#linear-function">linearly</a>, perhaps as the start of an exponential. We can see that all of them grow more or less linearly, perhaps as the start of an exponential. We can see that all of them grow more or less linearly, perhaps as the start of an exponential.<div class="list"><ul id="_51"><li id="_52">total dry mass (mass excluding <a href="chemistry#water">water</a>)</li><li id="_53"><a href="protein">protein</a> mass</li><li id="_54"><a href="cell#ribosomal-rna">rRNA</a> mass</li><li id="_55"><a href="dna#messenger-rna">mRNA</a> mass</li><li id="_56"><a href="dna">DNA</a> mass. The last label is not very visible on the plots, but we can deduce it from the source code.</li></ul></div>By grepping the title "Cell mass fractions" in the source code, we see the files:<div class="code" id="_57"><div><pre><code>models/ecoli/analysis/cohort/massFractionSummary.py
models/ecoli/analysis/multigen/massFractionSummary.py
models/ecoli/analysis/variant/massFractionSummary.py</code></pre></div></div>which must correspond to the different <code>massFractionSummary</code> plots throughout different levels of the hierarchy.</div><div class="p" id="_58">By reading <code>models/ecoli/analysis/variant/massFractionSummary.py</code> a little bit, we see that:<div class="list"><ul id="_59"><li id="_60">the plotting is done with <a href="software#matplotlib">Matplotlib</a>, hurray</li><li id="_61"><div class="p" id="_62">it is reading its data from files under <code>./out/manual/wildtype_000000/000000/generation_000000/000000/simOut/Mass/</code>, more precisely <code>./out/manual/wildtype_000000/000000/generation_000000/000000/simOut/Mass/columns/&lt;column-name&gt;/data</code>. They are binary files however.</div><div class="p" id="_63">Looking at the source for <code>wholecell/io/tablereader.py</code> shows that those are just a standard <a href="programming-language#numpy">NumPy</a> serialization mechanism. Maybe they should have used the <a href="computer#hierarchical-data-format">Hierarchical Data Format</a> instead.</div><div class="p" id="_64">We can also take this opportunity to try and find where the data is coming from. <code>Mass</code> from the <code>./out/manual/wildtype_000000/000000/generation_000000/000000/simOut/Mass/</code> looks like an ID, so we <a href="systems-programming#grep"><code>grep</code></a> that and we reach <code>models/ecoli/listeners/mass.py</code>.</div><div class="p" id="_65">From this we understand that all data that is to be saved from a simulation must be coming from listeners: likely nothing, or not much, is dumped by default, because otherwise it would take up too much disk space. You have to explicitly say what it is that you want to save via a listener that acts on each time step.</div></li></ul></div></div><div class="figure"><figure id="image-minimal-condition-mass-fraction-plot"><div class="float-wrap"><a href="https://upload.wikimedia.org/wikipedia/commons/9/94/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_mass_fraction_summary.svg"><img src="https://upload.wikimedia.org/wikipedia/commons/9/94/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_mass_fraction_summary.svg" loading="lazy" height="600"></a></div><figcaption><a href="#image-minimal-condition-mass-fraction-plot"><span class="caption-prefix">Figure 1. </span></a><div class="title">Minimal condition mass fraction plot</div>. <a href="https://commons.wikimedia.org/wiki/File:E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_mass_fraction_summary.svg">Source</a>. File name: <code>out/manual/plotOut/svg_plots/massFractionSummary.svg</code></figcaption></figure></div><div class="p" id="_66">More plot types will be explored at <a href="#time-series-run-variant">time series run variant</a>, where we will contrast two runs with different <a href="cell#growth-medium">growth mediums</a>.</div><div class="h" id="run-variants"><div class="notnav"><h2><a href="e-coli-whole-cell-model-by-covert-lab/run-variants"><span class="number">3. </span>Run variants</a></h2><span class="hover-meta"> <a class="split" href="e-coli-whole-cell-model-by-covert-lab/run-variants"></a><span class="metrics"><span class="wcntr"> Words: 928</span><span class="dcnt"> Articles: 2</span></span></span></div><nav class="h-nav h-nav-toplevel"><div class="nav"><a href="" class="u"> E. Coli Whole Cell Model by Covert Lab</a><a href="#_toc/run-variants" class="toc"></a><a href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab/run-variants"><img src="_obb/logo.svg" class="logo" alt=""> OurBigBook.com</a></div></nav></div><div class="p" id="_67">It would be boring if we could only simulate the same condition all the time, so let's have a look at the different <a href="calculus#boundary-condition">boundary conditions</a> that we can apply to the cell!</div><div class="p" id="_68">We are able to alter things like the composition of the external medium, and the genome of the bacteria, which will make the bacteria behave differently.</div><div class="p" id="_69">The variant selection is a bit cumbersome as we have to use indexes instead of names, but one you know what you are doing, it is fine.</div><div class="p" id="_70">Of course, genetic modification is limited only to experimentally known protein interactions due to the intractability of <a href="protein#computational-protein-folding">computational protein folding</a> and <a href="physics#computational-chemistry">computational chemistry</a> in general, solving those would bsai.</div><div class="h" id="default-run-variant"><div class="notnav"><h3><a href="e-coli-whole-cell-model-by-covert-lab/default-run-variant"><span class="number">3.1. </span>Default run variant</a></h3><span class="hover-meta"> <a class="split" href="e-coli-whole-cell-model-by-covert-lab/default-run-variant"></a><span class="metrics"><span class="wcntr"> Words: 80</span></span></span></div><nav class="h-nav h-nav-toplevel"><div class="nav"><a href="#run-variants" class="u"> Run variants</a><a href="#_toc/default-run-variant" class="toc"></a><a href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab/default-run-variant"><img src="_obb/logo.svg" class="logo" alt=""> OurBigBook.com</a></div></nav></div><div class="p" id="_71">The default run variant, if you don't pass any options, just has the minimal growth conditions set. What this means can be seen at <a href="#condition">condition</a>.</div><div class="p" id="_72">Notably, this implies a <a href="cell#growth-medium">growth medium</a> that includes <a href="chemistry#glucose">glucose</a> and <a href="chemistry#salt-chemistry">salt</a>. It also includes <a href="chemistry#oxygen">oxygen</a>, which is not strictly required, but greatly benefits cell growth, and is of course easier to have than not have as it is part of the atmosphere!</div><div class="p" id="_73">But the medium does not include <a href="protein#amino-acid">amino acids</a>, which the bacteria will have to produce by itself.</div><div class="h" id="time-series-run-variant"><div class="notnav"><h3><a href="e-coli-whole-cell-model-by-covert-lab/time-series-run-variant"><span class="number">3.2. </span>Time series run variant</a></h3><span class="hover-meta"> <a class="split" href="e-coli-whole-cell-model-by-covert-lab/time-series-run-variant"></a><span class="metrics"><span class="wcntr"> Words: 741</span></span></span></div><nav class="h-nav h-nav-toplevel"><div class="nav"><a href="#run-variants" class="u"> Run variants</a><a href="#_toc/time-series-run-variant" class="toc"></a><a href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab/time-series-run-variant"><img src="_obb/logo.svg" class="logo" alt=""> OurBigBook.com</a></div></nav></div><div class="p" id="_74">To modify the nutrients as a function of time, with To select a time series we can use something like:<div class="code" id="_75"><div><pre><code>python runscripts/manual/runSim.py --variant nutrientTimeSeries 25 25</code></pre></div></div>As mentioned in <code>python runscripts/manual/runSim.py --help</code>, <code>nutrientTimeSeries</code> is one of the choices from <a href="https://github.com/CovertLab/WholeCellEcoliRelease/blob/7e4cc9e57de76752df0f4e32eca95fb653ea64e4/models/ecoli/sim/variants/__init__.py#L57">github.com/CovertLab/WholeCellEcoliRelease/blob/7e4cc9e57de76752df0f4e32eca95fb653ea64e4/models/ecoli/sim/variants/__init__.py#L57</a></div><div class="p" id="_76"><code>25 25</code> means to start from index 25 and also end at 25, so running just one simulation. <code>25 27</code> would run 25 then 26 and then 27 for example.</div><div class="p" id="_77">The timeseries with index 25 is <code>reconstruction/ecoli/flat/condition/timeseries/000025_cut_aa.tsv</code> and contains<div class="code" id="_78"><div><pre><code>"time (units.s)" "nutrients"
0 "minimal_plus_amino_acids"
1200 "minimal"</code></pre></div></div>so we understand that it starts with extra <a href="protein#amino-acid">amino acids</a> in the medium, which benefit the cell, and half way through those are removed at time 1200s = 20 minutes. We would therefore expect the cell to start expressing amino acid production genes exactly at that point.</div><div class="p" id="_79"><code>nutrients</code> likely means <code>condition</code> in that file however, see bug report with <code>1 1</code> failing:  <a href="https://github.com/CovertLab/WholeCellEcoliRelease/issues/24">github.com/CovertLab/WholeCellEcoliRelease/issues/24</a></div><div class="p" id="_80">When we do this the simulation ends in:<div class="code" id="_81"><div><pre><code>Simulation finished:
 - Length: 0:34:23
 - Runtime: 0:08:03</code></pre></div></div>so we see that the doubling time was faster than the one with minimal conditions of <code>0:42:49</code>, which makes sense, since during the first 20 minutes the cell had extra <a href="protein#amino-acid">amino acid</a> nutrients at its disposal.</div><div class="p" id="_82">The output directory now contains simulation output data under <code>out/manual/nutrientTimeSeries_000025/</code>. Let's run analysis and plots for that:<div class="code" id="_83"><div><pre><code>python runscripts/manual/analysisVariant.py &amp;&amp;
python runscripts/manual/analysisCohort.py --variant 25 &amp;&amp;
python runscripts/manual/analysisMultigen.py --variant 25 &amp;&amp;
python runscripts/manual/analysisSingle.py --variant 25</code></pre></div></div></div><div class="p" id="_84">We can now compare the outputs of this run to the default <code>wildtype_000000</code> run from <a href="#install-and-first-run">Section 1. "Install and first run"</a>.</div><div class="list"><ul id="_85"><li id="_86"><div class="p" id="_87"><code>out/manual/plotOut/svg_plots/massFractionSummary.svg</code>: because we now have two variants in the same <code>out/</code> folder, <code>wildtype_000000</code> and <code>nutrientTimeSeries_000025</code>, we now see a side by side comparision of both on the same graph!</div><div class="p" id="_88">The run variant where we started with amino acids initially grows faster as expected, because the cell didn't have to make it's own amino acids, so growth is a bit more efficient.</div><div class="p" id="_89">Then, at 20 minutes, which is about 0.3 hours, we see that the cell starts growing a bit less fast as the slope of the curve decreases a bit, because we removed that free amino acid supply.</div><div class="figure"><figure id="image-minimal-condition-vs-amino-acid-cut-mass-fraction-plot"><div class="float-wrap"><a href="https://upload.wikimedia.org/wikipedia/commons/5/5f/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_acids_mass_fraction_summary.svg"><img src="https://upload.wikimedia.org/wikipedia/commons/5/5f/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_acids_mass_fraction_summary.svg" loading="lazy" height="600"></a></div><figcaption><a href="#image-minimal-condition-vs-amino-acid-cut-mass-fraction-plot"><span class="caption-prefix">Figure 2. </span></a><div class="title">Minimal condition vs amino acid cut mass fraction plot</div>. <a href="https://commons.wikimedia.org/wiki/File:E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_acids_mass_fraction_summary.svg">Source</a>. From file <code>out/manual/plotOut/svg_plots/massFractionSummary.svg</code>.</figcaption></figure></div></li></ul></div><div class="p" id="_90">The following plots from under <code>out/manual/wildtype_000000/000000/{generation_000000,nutrientTimeSeries_000025}/000000/plotOut/svg_plots</code> have been manually joined side-by-side with:<div class="code" id="_91"><div><pre><code>for f in out/manual/wildtype_000000/000000/generation_000000/000000/plotOut/svg_plots/*; do
  echo $f
  svg_stack.py \
    --direction h \
    out/manual/wildtype_000000/000000/generation_000000/000000/plotOut/svg_plots/$(basename $f) \
    out/manual/nutrientTimeSeries_000025/000000/generation_000000/000000/plotOut/svg_plots/$(basename $f) \
    &gt; tmp/$(basename $f)
done</code></pre></div></div></div><div class="figure"><figure id="image-amino-acid-counts" class="multiline"><div class="float-wrap"><a href="https://upload.wikimedia.org/wikipedia/commons/e/e1/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_acids_amino_acid_counts.svg"><img src="https://upload.wikimedia.org/wikipedia/commons/e/e1/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_acids_amino_acid_counts.svg" loading="lazy" height="800"></a></div><figcaption><a href="#image-amino-acid-counts"><span class="caption-prefix">Figure 3. </span></a><div class="title">Amino acid counts</div>. <a href="https://commons.wikimedia.org/wiki/File:E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_acids_amino_acid_counts.svg">Source</a>. <code>aaCounts.svg</code>:<div class="list"><ul id="_92"><li id="_93">default: quantities just increase</li><li id="_94">amino acid cut: there is an abrupt fall at 20 minutes when we cut off external supply, presumably because it takes some time for the cell to start producing its own</li></ul></div></figcaption></figure></div><div class="p" id="_95"><div class="figure"><figure id="image-external-exchange-fluxes-of-amino-acids" class="multiline"><div class="float-wrap"><a href="https://upload.wikimedia.org/wikipedia/commons/7/7e/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_exchange_fluxes_of_amino_acids.svg"><img src="https://upload.wikimedia.org/wikipedia/commons/7/7e/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_exchange_fluxes_of_amino_acids.svg" loading="lazy" height="800"></a></div><figcaption><a href="#image-external-exchange-fluxes-of-amino-acids"><span class="caption-prefix">Figure 4. </span></a><div class="title">External exchange fluxes of amino acids</div>. <a href="https://commons.wikimedia.org/wiki/File:E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_exchange_fluxes_of_amino_acids.svg">Source</a>. <code>aaExchangeFluxes.svg</code>:<div class="list"><ul id="_96"><li id="_97">default: no exchanges</li><li id="_98"><div class="p" id="_99">amino acid cut: for all graphs except <a href="protein#phenylalanine">phenylalanine</a> (PHE), either the cell was intaking the AA (negative flux), and that intake goes to 0 when the supply is cut, or the flux is always 0.</div><div class="p" id="_100">For PHE however, the flux is at all times, except shortly after the cut. Why? And why there was no excretion on the default conditions?</div></li></ul></div></figcaption></figure></div><div class="figure"><figure id="image-evaluation-time"><div class="float-wrap"><a href="https://upload.wikimedia.org/wikipedia/commons/d/d6/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_evaluation_time.svg"><img src="https://upload.wikimedia.org/wikipedia/commons/d/d6/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_evaluation_time.svg" loading="lazy" height="315"></a></div><figcaption><a href="#image-evaluation-time"><span class="caption-prefix">Figure 5. </span></a><div class="title">Evaluation time</div>. <a href="https://commons.wikimedia.org/wiki/File:E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_evaluation_time.svg">Source</a>. <code>evaluationTime.svg</code>: this has nothing to do with biology, but it is rather a <a href="software#profiling-computer-programming">profile</a> of the program runtime. We can see that the simulation gets slower and slower as time passes, presumably because there are more and more molecules to simulate.</figcaption></figure></div></div><div class="figure"><figure id="image-mrna-count-of-highly-expressed-mrnas" class="multiline"><div class="float-wrap"><a href="https://upload.wikimedia.org/wikipedia/commons/7/7a/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_mrna_count_of_highly_expressed_mRNAs.svg"><img src="https://upload.wikimedia.org/wikipedia/commons/7/7a/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_mrna_count_of_highly_expressed_mRNAs.svg" loading="lazy" height="800"></a></div><figcaption><a href="#image-mrna-count-of-highly-expressed-mrnas"><span class="caption-prefix">Figure 6. </span></a><div class="title">mRNA count of highly expressed mRNAs</div>. <a href="https://commons.wikimedia.org/wiki/File:E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_mrna_count_of_highly_expressed_mRNAs.svg">Source</a>. From file <code>expression_rna_03_high.svg</code>. Each of the entries is a <a href="dna#gene">gene</a> using the conventional gene naming convention of <code>xyzW</code>, e.g. here's the <a href="biology#biocyc">BioCyc</a> for the first entry, <code>tufA</code>: <a href="https://biocyc.org/gene?orgid=ECOLI&amp;id=EG11036">biocyc.org/gene?orgid=ECOLI&amp;id=EG11036</a>, which comments <div><blockquote id="_101">Elongation factor Tu (EF-Tu) is the most abundant protein in E. coli.</blockquote></div> and <div><blockquote id="_102">In E. coli, EF-Tu is encoded by two genes, tufA and tufB</blockquote></div>. What they seem to mean is that tufA and tufB are two similar molecules, either of which can make up the <a href="cell#ef-tu">EF-Tu</a> of the <a href="taxonomy#escherichia-coli">E. Coli</a>, which is an important part of <a href="cell#translation-biology">translation</a>.</figcaption></figure></div><div class="figure"><figure id="image-external-exchange-fluxes" class="multiline"><div class="float-wrap"><a href="https://upload.wikimedia.org/wikipedia/commons/4/49/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_exchange_fluxes.svg"><img src="https://upload.wikimedia.org/wikipedia/commons/4/49/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_exchange_fluxes.svg" loading="lazy" height="1000"></a></div><figcaption><a href="#image-external-exchange-fluxes"><span class="caption-prefix">Figure 7. </span></a><div class="title">External exchange fluxes</div>. <a href="https://commons.wikimedia.org/wiki/File:E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_exchange_fluxes.svg">Source</a>. <div class="p" id="_103"><code>mediaExcange.svg</code>: this one is similar to <code>aaExchangeFluxes.svg</code>, but it also tracks other substances. The color version makes it easier to squeeze more substances in a given space, but you lose the shape of curves a bit. The title seems reversed: red must be excretion, since that's where <a href="chemistry#glucose">glucose</a> (GLC) is.</div><div class="p" id="_104">The substances are different between the default and amino acid cut graphs, they seem to be the most exchanged substances. On the amino cut graph, first we see the cell intaking most (except <a href="protein#phenylalanine">phenylalanine</a>, which is excreted for some reason). When we cut amino acids, the uptake of course stops.</div></figcaption></figure></div><div class="h" id="other-run-variants"><div class="notnav"><h2><a href="e-coli-whole-cell-model-by-covert-lab/other-run-variants"><span class="number">4. </span>Other run variants</a></h2><span class="hover-meta"> <a class="split" href="e-coli-whole-cell-model-by-covert-lab/other-run-variants"></a><span class="metrics"><span class="wcntr"> Words: 75</span></span></span></div><nav class="h-nav h-nav-toplevel"><div class="nav"><a href="" class="u"> E. Coli Whole Cell Model by Covert Lab</a><a href="#_toc/other-run-variants" class="toc"></a><a href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab/other-run-variants"><img src="_obb/logo.svg" class="logo" alt=""> OurBigBook.com</a></div></nav></div><div class="p" id="_105">Besides <a href="#time-series-run-variant">time series run variants</a>, conditions can also be selected directly without a time series as in:<div class="code" id="_106"><div><pre><code>python runscripts/manual/runSim.py --variant condition 1 1</code></pre></div></div>which select row indices from <code>reconstruction/ecoli/flat/condition/condition_defs.tsv</code>. The above <code>1 1</code> would mean the second line of that file which starts with:<div class="code" id="_107"><div><pre><code>"condition" "nutrients" "genotype perturbations" "doubling time (units.min)" "active TFs"
"basal" "minimal" {} 44.0 []
"no_oxygen" "minimal_minus_oxygen" {} 100.0 []
"with_aa" "minimal_plus_amino_acids" {} 25.0 ["CPLX-125", "MONOMER0-162", "CPLX0-7671", "CPLX0-228", "MONOMER0-155"]</code></pre></div></div>so <code>1</code> means <code>no_oxygen</code>.</div><div class="h" id="source-code-overview"><div class="notnav"><h2><a href="e-coli-whole-cell-model-by-covert-lab/source-code-overview"><span class="number">5. </span>Source code overview</a></h2><span class="hover-meta"> <a class="split" href="e-coli-whole-cell-model-by-covert-lab/source-code-overview"></a><span class="metrics"><span class="wcntr"> Words: 1k</span><span class="dcnt"> Articles: 1</span></span></span></div><nav class="h-nav h-nav-toplevel"><div class="nav"><a href="" class="u"> E. Coli Whole Cell Model by Covert Lab</a><a href="#_toc/source-code-overview" class="toc"></a><a href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab/source-code-overview"><img src="_obb/logo.svg" class="logo" alt=""> OurBigBook.com</a></div></nav></div><div class="p" id="_108">The key model database is located in the source code at <code>reconstruction/ecoli/flat</code>.</div><div class="p" id="_109">Let's try to understand some interesting looking, with a special focus on our understanding of the tiny <a href="taxonomy#e-coli-k-12-mg1655-operon-thrlabc">E. Coli K-12 MG1655 operon thrLABC</a> part of the metabolism, which we have well understood at <a href="taxonomy#e-coli-k-12-mg1655-operon-thrlabc">Section "E. Coli K-12 MG1655 operon thrLABC"</a>.</div><div class="p" id="_110">We'll realize that a lot of data and IDs come from/match <a href="biology#biocyc">BioCyc</a> quite closely.</div><div class="list"><ul id="_111"><li id="_112"><div class="p" id="_113"><code>reconstruction/ecoli/flat/compartments.tsv</code> contains <a href="cell#cellular-compartment">cellular compartment</a> information:<div class="code" id="_114"><div><pre><code>"abbrev" "id"
"n" "CCO-BAC-NUCLEOID"
"j" "CCO-CELL-PROJECTION"
"w" "CCO-CW-BAC-NEG"
"c" "CCO-CYTOSOL"
"e" "CCO-EXTRACELLULAR"
"m" "CCO-MEMBRANE"
"o" "CCO-OUTER-MEM"
"p" "CCO-PERI-BAC"
"l" "CCO-PILUS"
"i" "CCO-PM-BAC-NEG"</code></pre></div></div></div><div class="list"><ul id="_115"><li id="_116"><code>CCO</code>: "Celular COmpartment"</li><li id="_117"><code>BAC-NUCLEOID</code>: <a href="cell#nucleoid">nucleoid</a></li><li id="_118"><code>CELL-PROJECTION</code>: <a href="cell#cell-projection">cell projection</a></li><li id="_119"><code>CW-BAC-NEG</code>: TODO confirm: <a href="cell#cell-wall">cell wall</a> (of a <a href="taxonomy#gram-negative-bacteria">Gram-negative bacteria</a>)</li><li id="_120"><code>CYTOSOL</code>: <a href="cell#cytosol">cytosol</a></li><li id="_121"><code>EXTRACELLULAR</code>: outside the cell</li><li id="_122"><code>MEMBRANE</code>: <a href="cell#cell-membrane">cell membrane</a></li><li id="_123"><code>OUTER-MEM</code>: <a href="taxonomy#bacterial-outer-membrane">bacterial outer membrane</a></li><li id="_124"><code>PERI-BAC</code>: <a href="taxonomy#periplasm">periplasm</a></li><li id="_125"><code>PILUS</code>: <a href="cell#pilus">pilus</a></li><li id="_126"><code>PM-BAC-NEG</code>: TODO: <a href="cell#organelle">plasma membrane</a>, but that is the same as <a href="cell#cell-membrane">cell membrane</a> no?</li></ul></div></li><li id="_127"><code>reconstruction/ecoli/flat/promoters.tsv</code> contains <a href="dna#promoter-genetics">promoter</a> information. Simple file, sample lines:<div class="code" id="_128"><div><pre><code>"position" "direction" "id" "name"
148 "+" "PM00249" "thrLp"</code></pre></div></div>corresponds to <a href="taxonomy#e-coli-k-12-mg1655-promoter-thrlp">E. Coli K-12 MG1655 promoter thrLp</a>, which starts as position 148.</li><li id="_129"><code>reconstruction/ecoli/flat/proteins.tsv</code> contains <a href="protein">protein</a> information. Sample line corresponding to <a href="taxonomy#e-coli-k-12-mg1655-gene-thra">e. Coli K-12 MG1655 gene thrA</a>:<div class="code" id="_130"><div><pre><code>"aaCount" "name" "seq" "comments" "codingRnaSeq" "mw" "location" "rnaId" "id" "geneId"
[91, 46, 38, 44, 12, 53, 30, 63, 14, 46, 89, 34, 23, 30, 29, 51, 34, 4, 20, 0, 69] "ThrA" "MRVL..." "Location information from Ecocyc dump." "AUGCGAGUGUUG..." [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 89103.51099999998, 0.0, 0.0, 0.0, 0.0] ["c"] "EG10998_RNA" "ASPKINIHOMOSERDEHYDROGI-MONOMER" "EG10998"</code></pre></div></div>so we understand that:<div class="list"><ul id="_131"><li id="_132"><code>aaCount</code>: <a href="protein#amino-acid">amino acid</a> count, how many of each of the 20 <a href="protein#proteinogenic-amino-acid">proteinogenic amino acid</a> are there</li><li id="_133"><code>seq</code>: full sequence, using the single letter abbreviation of the <a href="protein#proteinogenic-amino-acid">proteinogenic amino acids</a></li><li id="_134"><code>mw</code>; molecular weight? The 11 components appear to be given at <code>reconstruction/ecoli/flat/scripts/unifyBulkFiles.py</code>:<div class="code" id="_135"><div><pre><code>molecular_weight_keys = [
  '23srRNA',
  '16srRNA',
  '5srRNA',
  'tRNA',
  'mRNA',
  'miscRNA',
  'protein',
  'metabolite',
  'water',
  'DNA',
  'RNA' # nonspecific RNA
  ]</code></pre></div></div>so they simply classify the weight? Presumably this exists for complexes that have multiple classes?<div class="list"><ul id="_136"><li id="_137"><code>23srRNA</code>, <code>16srRNA</code>, <code>5srRNA</code> are the three structural <a href="dna#rna">RNAs</a> present in the <a href="cell#ribosome">ribosome</a>: <a href="cell#23s-ribosomal-rna">23S ribosomal RNA</a>, <a href="cell#16s-ribosomal-rna">16S ribosomal RNA</a>, <a href="cell#5s-ribosomal-rna">5S ribosomal RNA</a>, all others are obvious:</li><li id="_138"><a href="cell#transfer-rna">tRNA</a></li><li id="_139"><a href="dna#messenger-rna">mRNA</a></li><li id="_140"><a href="protein">protein</a>. This is the seventh class, and this enzyme only contains mass in this class as expected.</li><li id="_141"><a href="molecular-biology#metabolite">metabolite</a></li><li id="_142"><a href="chemistry#water">water</a></li><li id="_143"><a href="dna">DNA</a></li><li id="_144"><a href="dna#rna">RNA</a>: TODO <code>rna</code> vs <code>miscRNA</code></li></ul></div></li><li id="_145"><code>location</code>: <a href="cell#cellular-compartment">cell compartment</a> where the protein is present, <code>c</code> defined at <code>reconstruction/ecoli/flat/compartments.tsv</code> as <a href="cell#cytoplasm">cytoplasm</a>, as expected for something that will make an <a href="protein#amino-acid">amino acid</a></li></ul></div></li><li id="_146"><div class="p" id="_147"><code>reconstruction/ecoli/flat/rnas.tsv</code>: TODO vs <code>transcriptionUnits.tsv</code>. Sample lines:<div class="code" id="_148"><div><pre><code>"halfLife" "name" "seq" "type" "modifiedForms" "monomerId" "comments" "mw" "location" "ntCount" "id" "geneId" "microarray expression"
174.0 "ThrA [RNA]" "AUGCGAGUGUUG..." "mRNA" [] "ASPKINIHOMOSERDEHYDROGI-MONOMER" "" [0.0, 0.0, 0.0, 0.0, 790935.00399999996, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0] ["c"] [553, 615, 692, 603] "EG10998_RNA" "EG10998" 0.0005264904</code></pre></div></div></div><div class="list"><ul id="_149"><li id="_150"><code>halfLife</code>: <a href="particle-physics#half-life">half-life</a></li><li id="_151"><code>mw</code>: molecular weight, same as in <code>reconstruction/ecoli/flat/proteins.tsv</code>. This <a href="quantum-mechanics#molecule">molecule</a> only have weight in the <code>mRNA</code> class, as expected, as it just codes for a protein</li><li id="_152"><code>location</code>: same as in <code>reconstruction/ecoli/flat/proteins.tsv</code></li><li id="_153"><code>ntCount</code>: <a href="dna#nucleotide">nucleotide</a> count for each of the ATGC</li><li id="_154"><code>microarray expression</code>: presumably refers to <a href="dna#dna-microarray">DNA microarray</a> for <a href="dna#gene-expression-profiling">gene expression profiling</a>, but what measure exactly?</li></ul></div></li><li id="_155"><code>reconstruction/ecoli/flat/sequence.fasta</code>: <a href="biology#fasta-format">FASTA</a> <a href="dna">DNA</a> sequence, first two lines:<div class="code" id="_156"><div><pre><code>&gt;E. coli K-12 MG1655 U00096.2 (1 to 4639675 = 4639675 bp)
AGCTTTTCATTCTGACTGCAACGGGCAATATGTCTCTGTGTGGATTAAAAAAAGAGTGTCTGATAGCAGCTTCTG</code></pre></div></div></li><li id="_157"><div class="p" id="_158"><code>reconstruction/ecoli/flat/transcriptionUnits.tsv</code>: <a href="dna#transcription-unit">transcription units</a>. We can observe for example the two different transcription units of the <a href="taxonomy#e-coli-k-12-mg1655-operon-thrlabc">E. Coli K-12 MG1655 operon thrLABC</a> in the lines:<div class="code" id="_159"><div><pre><code>"expression_rate" "direction" "right" "terminator_id"  "name"    "promoter_id" "degradation_rate" "id"       "gene_id"                                   "left"
0.0               "f"         310     ["TERM0-1059"]   "thrL"    "PM00249"     0.198905992329492 "TU0-42486" ["EG11277"]                                  148
657.057317358791  "f"         5022    ["TERM_WC-2174"] "thrLABC" "PM00249"     0.231049060186648 "TU00178"   ["EG10998", "EG10999", "EG11000", "EG11277"] 148</code></pre></div></div></div><div class="list"><ul id="_160"><li id="_161"><code>promoter_id</code>: matches promoter id in <code>reconstruction/ecoli/flat/promoters.tsv</code></li><li id="_162"><code>gene_id</code>: matches id in <code>reconstruction/ecoli/flat/genes.tsv</code></li><li id="_163"><code>id</code>: matches exactly those used in <a href="biology#biocyc">BioCyc</a>, which is quite nice, might be more or less standardized:<div class="list"><ul id="_164"><li id="_165"><a href="https://biocyc.org/ECOLI/NEW-IMAGE?object=TU0-42486">biocyc.org/ECOLI/NEW-IMAGE?object=TU0-42486</a></li><li id="_166"><a href="https://biocyc.org/ECOLI/NEW-IMAGE?type=OPERON&amp;object=TU00178">biocyc.org/ECOLI/NEW-IMAGE?type=OPERON&amp;object=TU00178</a></li></ul></div></li></ul></div></li><li id="_167"><code>reconstruction/ecoli/flat/genes.tsv</code><div class="code" id="_168"><div><pre><code>"length" "name"                      "seq"             "rnaId"      "coordinate" "direction" "symbol" "type" "id"      "monomerId"
66       "thr operon leader peptide" "ATGAAACGCATT..." "EG11277_RNA" 189         "+"         "thrL"   "mRNA" "EG11277" "EG11277-MONOMER"
2463     "ThrA"                      "ATGCGAGTGTTG"    "EG10998_RNA" 336         "+"         "thrA"   "mRNA" "EG10998" "ASPKINIHOMOSERDEHYDROGI-MONOMER"</code></pre></div></div></li><li id="_169"><div class="p" id="_170"><code>reconstruction/ecoli/flat/metabolites.tsv</code> contains <a href="molecular-biology#metabolite">metabolite</a> information. Sample lines:<div class="code" id="_171"><div><pre><code>"id"                       "mw7.2" "location"
"HOMO-SER"                 119.12  ["n", "j", "w", "c", "e", "m", "o", "p", "l", "i"]
"L-ASPARTATE-SEMIALDEHYDE" 117.104 ["n", "j", "w", "c", "e", "m", "o", "p", "l", "i"]</code></pre></div></div>In the case of the enzyme thrA, one of the two reactions it catalyzes is "L-aspartate 4-semialdehyde" into "Homoserine".</div><div class="p" id="_172">Starting from the enzyme page: <a href="https://biocyc.org/gene?orgid=ECOLI&amp;id=EG10998">biocyc.org/gene?orgid=ECOLI&amp;id=EG10998</a> we reach the reaction page: <a href="https://biocyc.org/ECOLI/NEW-IMAGE?type=REACTION&amp;object=HOMOSERDEHYDROG-RXN">biocyc.org/ECOLI/NEW-IMAGE?type=REACTION&amp;object=HOMOSERDEHYDROG-RXN</a> which has reaction ID <code>HOMOSERDEHYDROG-RXN</code>, and that page which clarifies the IDs:<div class="list"><ul id="_173"><li id="_174"><a href="https://biocyc.org/compound?orgid=ECOLI&amp;id=L-ASPARTATE-SEMIALDEHYDE:">biocyc.org/compound?orgid=ECOLI&amp;id=L-ASPARTATE-SEMIALDEHYDE:</a> "L-aspartate 4-semialdehyde" has ID <code>L-ASPARTATE-SEMIALDEHYDE</code></li><li id="_175"><a href="https://biocyc.org/compound?orgid=ECOLI&amp;id=HOMO-SER:">biocyc.org/compound?orgid=ECOLI&amp;id=HOMO-SER:</a> "Homoserine" has ID <code>HOMO-SER</code></li></ul></div>so these are the compounds that we care about.</div></li><li id="_176"><div class="p" id="_177"><code>reconstruction/ecoli/flat/reactions.tsv</code> contains <a href="chemistry#chemical-reaction">chemical reaction</a> information. Sample lines:<div class="code" id="_178"><div><pre><code>"reaction id" "stoichiometry" "is reversible" "catalyzed by"

"HOMOSERDEHYDROG-RXN-HOMO-SER/NAD//L-ASPARTATE-SEMIALDEHYDE/NADH/PROTON.51."
  {"NADH[c]": -1, "PROTON[c]": -1, "HOMO-SER[c]": 1, "L-ASPARTATE-SEMIALDEHYDE[c]": -1, "NAD[c]": 1}
  false
  ["ASPKINIIHOMOSERDEHYDROGII-CPLX", "ASPKINIHOMOSERDEHYDROGI-CPLX"]

"HOMOSERDEHYDROG-RXN-HOMO-SER/NADP//L-ASPARTATE-SEMIALDEHYDE/NADPH/PROTON.53."
  {"NADPH[c]": -1, "NADP[c]": 1, "PROTON[c]": -1, "L-ASPARTATE-SEMIALDEHYDE[c]": -1, "HOMO-SER[c]": 1
  false
  ["ASPKINIIHOMOSERDEHYDROGII-CPLX", "ASPKINIHOMOSERDEHYDROGI-CPLX"]</code></pre></div></div></div><div class="list"><ul id="_179"><li id="_180"><code>catalized by</code>: here we see <code>ASPKINIHOMOSERDEHYDROGI-CPLX</code>, which we can guess is a <a href="protein#protein-complex">protein complex</a> made out of <code>ASPKINIHOMOSERDEHYDROGI-MONOMER</code>, which is the ID for the <code>thrA</code> we care about! This is confirmed in <code>complexationReactions.tsv</code>.</li></ul></div></li><li id="_181"><code>reconstruction/ecoli/flat/complexationReactions.tsv</code> contains information about <a href="chemistry#chemical-reaction">chemical reactions</a> that produce <a href="protein#protein-complex">protein complexes</a>:<div class="code" id="_182"><div><pre><code>"process" "stoichiometry" "id" "dir"
"complexation"
  [
    {
      "molecule": "ASPKINIHOMOSERDEHYDROGI-CPLX",
      "coeff": 1,
      "type": "proteincomplex",
      "location": "c",
      "form": "mature"
    },
    {
      "molecule": "ASPKINIHOMOSERDEHYDROGI-MONOMER",
      "coeff": -4,
      "type": "proteinmonomer",
      "location": "c",
      "form": "mature"
    }
  ]
"ASPKINIHOMOSERDEHYDROGI-CPLX_RXN"
1</code></pre></div></div>The <code>coeff</code> is how many monomers need to get together for form the final complex. This can be seen from the Summary section of <a href="https://ecocyc.org/gene?orgid=ECOLI&amp;id=ASPKINIHOMOSERDEHYDROGI-MONOMER">ecocyc.org/gene?orgid=ECOLI&amp;id=ASPKINIHOMOSERDEHYDROGI-MONOMER</a>:<div><blockquote id="_183">Aspartate kinase I / homoserine dehydrogenase I comprises a dimer of ThrA dimers. Although the dimeric form is catalytically active, the binding equilibrium dramatically favors the tetrameric form. The aspartate kinase and homoserine dehydrogenase activities of each ThrA monomer are catalyzed by independent domains connected by a linker region.</blockquote></div>Fantastic literature summary! Can't find that in database form there however.</li><li id="_184"><code>reconstruction/ecoli/flat/proteinComplexes.tsv</code> contains <a href="protein#protein-complex">protein complex</a> information:<div class="code" id="_185"><div><pre><code>"name" "comments" "mw" "location" "reactionId" "id"
"aspartate kinase / homoserine dehydrogenase"
""
[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 356414.04399999994, 0.0, 0.0, 0.0, 0.0]
["c"]
"ASPKINIHOMOSERDEHYDROGI-CPLX_RXN"
"ASPKINIHOMOSERDEHYDROGI-CPLX"</code></pre></div></div></li><li id="_186"><code>reconstruction/ecoli/flat/protein_half_lives.tsv</code> contains the <a href="particle-physics#half-life">half-life</a> of <a href="protein">proteins</a>. Very few proteins are listed however for some reason.</li><li id="_187"><code>reconstruction/ecoli/flat/tfIds.csv</code>: <a href="dna#transcription-factor">transcription factors</a> information:<div class="code" id="_188"><div><pre><code>"TF"   "geneId"  "oneComponentId"  "twoComponentId" "nonMetaboliteBindingId" "activeId" "notes"
"arcA" "EG10061" "PHOSPHO-ARCA"    "PHOSPHO-ARCA"
"fnr"  "EG10325" "FNR-4FE-4S-CPLX" "FNR-4FE-4S-CPLX"
"dksA" "EG10230"</code></pre></div></div></li></ul></div><div class="h" id="condition"><div class="notnav"><h3><a href="e-coli-whole-cell-model-by-covert-lab/condition"><span class="number">5.1. </span>Condition</a></h3><span class="hover-meta"> <a class="split" href="e-coli-whole-cell-model-by-covert-lab/condition"></a><span class="metrics"><span class="wcntr"> Words: 471</span></span></span></div><nav class="h-nav h-nav-toplevel"><div class="nav"><a href="#source-code-overview" class="u"> Source code overview</a><a href="#_toc/condition" class="toc"></a><a href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab/condition"><img src="_obb/logo.svg" class="logo" alt=""> OurBigBook.com</a></div></nav></div><div class="list"><ul id="_189"><li id="_190"><div class="p" id="_191"><code>reconstruction/ecoli/flat/condition/nutrient/minimal.tsv</code> contains the nutrients in a minimal environment in which the cell survives:<div class="code" id="_192"><div><pre><code>"molecule id" "lower bound (units.mmol / units.g / units.h)" "upper bound (units.mmol / units.g / units.h)"
"ADP[c]" 3.15 3.15
"PI[c]" 3.15 3.15
"PROTON[c]" 3.15 3.15
"GLC[p]" NaN 20
"OXYGEN-MOLECULE[p]" NaN NaN
"AMMONIUM[c]" NaN NaN
"PI[p]" NaN NaN
"K+[p]" NaN NaN
"SULFATE[p]" NaN NaN
"FE+2[p]" NaN NaN
"CA+2[p]" NaN NaN
"CL-[p]" NaN NaN
"CO+2[p]" NaN NaN
"MG+2[p]" NaN NaN
"MN+2[p]" NaN NaN
"NI+2[p]" NaN NaN
"ZN+2[p]" NaN NaN
"WATER[p]" NaN NaN
"CARBON-DIOXIDE[p]" NaN NaN
"CPD0-1958[p]" NaN NaN
"L-SELENOCYSTEINE[c]" NaN NaN
"GLC-D-LACTONE[c]" NaN NaN
"CYTOSINE[c]" NaN NaN</code></pre></div></div>If we compare that to <code>reconstruction/ecoli/flat/condition/nutrient/minimal_plus_amino_acids.tsv</code>, we see that it adds the 20 <a href="protein#amino-acid">amino acids</a> on top of the minimal condition:<div class="code" id="_193"><div><pre><code>"L-ALPHA-ALANINE[p]" NaN NaN
"ARG[p]" NaN NaN
"ASN[p]" NaN NaN
"L-ASPARTATE[p]" NaN NaN
"CYS[p]" NaN NaN
"GLT[p]" NaN NaN
"GLN[p]" NaN NaN
"GLY[p]" NaN NaN
"HIS[p]" NaN NaN
"ILE[p]" NaN NaN
"LEU[p]" NaN NaN
"LYS[p]" NaN NaN
"MET[p]" NaN NaN
"PHE[p]" NaN NaN
"PRO[p]" NaN NaN
"SER[p]" NaN NaN
"THR[p]" NaN NaN
"TRP[p]" NaN NaN
"TYR[p]" NaN NaN
"L-SELENOCYSTEINE[c]" NaN NaN
"VAL[p]" NaN NaN</code></pre></div></div>so we guess that <code>NaN</code> in the <code>upper mound</code> likely means infinite.</div><div class="p" id="_194">We can try to understand the less obvious ones:<div class="list"><ul id="_195"><li id="_196"><code>ADP</code>: TODO</li><li id="_197"><code>PI</code>: TODO</li><li id="_198"><code>PROTON[c]</code>: presumably a measure of <a href="chemistry#ph">pH</a></li><li id="_199"><code>GLC[p]</code>: <a href="chemistry#glucose">glucose</a>, this can be seen by comparing <code>minimal.tsv</code> with <code>minimal_no_glucose.tsv</code></li><li id="_200"><code>AMMONIUM</code>: <a href="chemistry#ammonium">ammonium</a>. This appears to be the primary source of <a href="chemistry#nitrogen">nitrogen</a> <a href="chemistry#atom">atoms</a> for producing <a href="protein#amino-acid">amino acids</a>.</li><li id="_201"><code>CYTOSINE[c]</code>: hmmm, why is external <a href="dna#cytosine">cytosine</a> needed? Weird.</li></ul></div></div></li><li id="_202"><div class="p" id="_203"><code>reconstruction/ecoli/flat/reconstruction/ecoli/flat/condition/timeseries/</code> contains sequences of conditions for each time. For example:<div class="list"><ul id="_204"><li id="_205"><code>reconstruction/ecoli/flat/reconstruction/ecoli/flat/condition/timeseries/000000_basal.tsv</code> contains:<div class="code" id="_206"><div><pre><code>"time (units.s)" "nutrients"
0 "minimal"</code></pre></div></div>which means just using <code>reconstruction/ecoli/flat/condition/nutrient/minimal.tsv</code> until infinity. That is the default one used by <code>runSim.py</code>, as can be seen from <code>./out/manual/wildtype_000000/000000/generation_000000/000000/simOut/Environment/attributes/nutrientTimeSeriesLabel</code> which contains just <code>000000_basal</code>.</li><li id="_207"><code>reconstruction/ecoli/flat/reconstruction/ecoli/flat/condition/timeseries/000001_cut_glucose.tsv</code> is more interesting and contains:<div class="code" id="_208"><div><pre><code>"time (units.s)" "nutrients"
0 "minimal"
1200 "minimal_no_glucose"</code></pre></div></div>so we see that this will shift the conditions half-way to a condition that will eventually kill the bacteria because it will run out of <a href="chemistry#glucose">glucose</a> and thus energy!</li></ul></div></div><div class="p" id="_209">Timeseries can be selected with <code>--variant nutrientTimeSeries X Y</code>, see also: <a href="#run-variants">run variants</a>.</div><div class="p" id="_210">We can use that variant with:<div class="code" id="_211"><div><pre><code>VARIANT="condition" FIRST_VARIANT_INDEX=1 LAST_VARIANT_INDEX=1 python runscripts/manual/runSim.py</code></pre></div></div></div></li><li id="_212"><div class="p" id="_213"><code>reconstruction/ecoli/flat/condition/condition_defs.tsv</code> contains lines of form:<div class="code" id="_214"><div><pre><code>"condition" "nutrients"                "genotype perturbations" "doubling time (units.min)" "active TFs"
"basal"     "minimal"                  {}                       44.0                        []
"no_oxygen" "minimal_minus_oxygen"     {}                       100.0                       []
"with_aa"   "minimal_plus_amino_acids" {}                       25.0                        ["CPLX-125", "MONOMER0-162", "CPLX0-7671", "CPLX0-228", "MONOMER0-155"]</code></pre></div></div></div><div class="list"><ul id="_215"><li id="_216"><code>condition</code> refers to entries in <code>reconstruction/ecoli/flat/condition/condition_defs.tsv</code></li><li id="_217"><code>nutrients</code> refers to entries under <code>reconstruction/ecoli/flat/condition/nutrient/</code>, e.g. <code>reconstruction/ecoli/flat/condition/nutrient/minimal.tsv</code> or <code>reconstruction/ecoli/flat/condition/nutrient/minimal_plus_amino_acids.tsv</code></li><li id="_218"><code>genotype perturbations</code>: there aren't any in the file, but this suggests that genotype modifications can also be incorporated here</li><li id="_219"><code>doubling time</code>: TODO experimental data? Because this should be a simulation output, right? Or do they cheat and fix doubling by time?</li><li id="_220"><code>active TFs</code>: this suggests that they are cheating <a href="dna#transcription-factor">transcription factors</a> here, as those would ideally be functions of other more basic inputs</li></ul></div></li></ul></div><div class="h" id="model-validation"><div class="notnav"><h2><a href="e-coli-whole-cell-model-by-covert-lab/model-validation"><span class="number">6. </span>Model validation</a></h2><span class="hover-meta"> <a class="split" href="e-coli-whole-cell-model-by-covert-lab/model-validation"></a><span class="metrics"><span class="wcntr"> Words: 5</span></span></span></div><nav class="h-nav h-nav-toplevel"><div class="nav"><a href="" class="u"> E. Coli Whole Cell Model by Covert Lab</a><a href="#_toc/model-validation" class="toc"></a><a href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab/model-validation"><img src="_obb/logo.svg" class="logo" alt=""> OurBigBook.com</a></div></nav></div><div class="p" id="_221">TODO compare with actual datasetes.</div><div class="h" id="publications"><div class="notnav"><h2><a href="e-coli-whole-cell-model-by-covert-lab/publications"><span class="number">7. </span>Publications</a></h2><span class="hover-meta"> <a class="split" href="e-coli-whole-cell-model-by-covert-lab/publications"></a><span class="metrics"><span class="wcntr"> Words: 78</span></span></span></div><nav class="h-nav h-nav-toplevel"><div class="nav"><a href="" class="u"> E. Coli Whole Cell Model by Covert Lab</a><a href="#_toc/publications" class="toc"></a><a href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab/publications"><img src="_obb/logo.svg" class="logo" alt=""> OurBigBook.com</a></div></nav></div><div class="p" id="_222">Unfortunately, due to lack of <a href="cirosantilli-com#one-page-to-rule-them-all">one page to rule them all</a>, the <a href="https://github.com/CovertLab/WholeCellEcoliRelease/blob/7e4cc9e57de76752df0f4e32eca95fb653ea64e4/docs/README.md#relevant-papers">on-Git tree publication list is meager</a>, some of the most relevant ones seems to be:<div class="list"><ul id="_223"><li id="_224">2021 <a href="education#open-access">open access</a> <a href="education#review-article">review paper</a>: <a href="https://journals.asm.org/doi/full/10.1128/ecosalplus.ESP-0001-2020">journals.asm.org/doi/full/10.1128/ecosalplus.ESP-0001-2020</a> "The E. coli Whole-Cell Modeling Project". They should just past that stuff in a <a href="software#readme">README</a> :-) The article mentions that it is a follow up to the previous <a href="taxonomy#m-genitalium-whole-cell-model-by-covert-lab">M. genitalium whole cell model by Covert lab</a>. Only 43% of known genes modelled at this point however, a shame.</li><li id="_225">2020 under <a href="education#science-journal">Science</a> <a href="education#paywall">paywall</a>: <a href="https://www.science.org/doi/10.1126/science.aav3751">www.science.org/doi/10.1126/science.aav3751</a> "Simultaneous cross-evaluation of heterogeneous E. coli datasets via mechanistic simulation"</li></ul></div></div><h2 id="_ancestors"><a href="#_ancestors"><span class="fa-solid-900 icon"></span> Ancestors <span class="meta">(10)</span></a></h2><div class="list"><ol><li><a href="taxonomy#e-coli-whole-cell-simulation">E. Coli whole cell simulation</a></li><li><a href="taxonomy#escherichia-coli">Escherichia coli</a></li><li><a href="taxonomy#list-of-bacteria">List of bacteria</a></li><li><a href="taxonomy#bacteria">Bacteria</a></li><li><a href="taxonomy#species">Species</a></li><li><a href="taxonomy">Taxonomy</a></li><li><a href="biology">Biology</a></li><li><a href="science#natural-science">Natural science</a></li><li><a href="science">Science</a></li><li><a href="."><span title="Home" class="fa-solid-900 icon"></span> Home</a></li></ol></div><h2 id="_incoming-links"><a href="#_incoming-links"><span title="Incoming links" class="fa-solid-900 icon"></span> Incoming links <span class="meta">(6)</span></a></h2><div class="list"><ul><li><a href="articles">The best articles by Ciro Santilli</a></li><li><a href="university#covert-lab">Covert Lab</a></li><li><a href="particle-physics#half-life">Half-life</a></li><li><a href="how-to-teach#exams-and-homework-are-useless-only-projects-matter">Exams and homework are useless, only projects matter</a></li><li><a href="taxonomy#m-genitalium-whole-cell-model-by-covert-lab">M. genitalium whole cell model by Covert lab</a></li><li><a href="systems-programming#molecular-biology-feels-like-systems-programming">Molecular biology feels like systems programming</a></li></ul></div>
</main>
<footer>
<div>Powered by <a href="https://docs.ourbigbook.com">OurBigBook</a></div>
<div>License: <a href="https://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA 4.0</a> unless noted</div>
<div><a href="https://github.com/cirosantilli/cirosantilli.github.io/issues">Suggestions and corrections</a></div>
<div><a href="contact">Contact Ciro Santilli</a></div>
<div><a href="_dir">Website source code</a></div>
<div><a href="https://github.com/cirosantilli/cirosantilli.github.io">Website source code on GitHub</a></div>
<div><a href="_file/e-coli-whole-cell-model-by-covert-lab.bigb">Source code for this page: e-coli-whole-cell-model-by-covert-lab.bigb</a></div>
<div><a href="https://github.com/cirosantilli/cirosantilli.github.io/blob/f7dac29d5224c15015eee520ccee15a3adb07b98/e-coli-whole-cell-model-by-covert-lab.bigb">Source code for this page on GitHub</a></div>
<div>Cite with: <a href="https://zenodo.org/badge/latestdoi/16453261">this DOI</a></div>
<div><img src="https://raw.githubusercontent.com/cirosantilli/media/master/ID_photo_of_Ciro_Santilli_taken_in_2013_left_eye_200_100.jpg" loading="lazy" alt="ID photo of Ciro Santilli taken in 2013 right eye"></div>
</footer>
<script>
window.ourbigbook_split_headers = true;
window.ourbigbook_html_x_extension = false;
window.ourbigbook_redirect_prefix = "e-coli-whole-cell-model-by-covert-lab/";
</script>
<script src="_obb/dist/ourbigbook_runtime.js"></script><script>ourbigbook_runtime.ourbigbook_runtime()</script><script src="_raw/main.js"></script>
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
ga('create', 'UA-47867706-1', 'auto');
ga('send', 'pageview');
</script>
<script async src="https://www.googletagmanager.com/gtag/js?id=G-DEE2HEJW9X"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'G-DEE2HEJW9X');
</script>
<script src="https://giscus.app/client.js"
        data-repo="cirosantilli/cirosantilli.github.io"
        data-repo-id="MDEwOlJlcG9zaXRvcnkxNjQ1MzI2MQ=="
        data-category="giscus"
        data-category-id="DIC_kwDOAPsOjc4CZ6zZ"
        data-mapping="pathname"
        data-strict="0"
        data-reactions-enabled="1"
        data-emit-metadata="0"
        data-input-position="bottom"
        data-theme="dark_high_contrast"
        data-lang="en"
        crossorigin="anonymous"
        async>
</script>
</body>
</html>
