<!doctype html>
<html lang=en>
<head>
<meta charset=utf-8>
<title>E. Coli Whole Cell Model by Covert Lab / Time series run variant - Ciro Santilli</title>
<meta property="og:title" content="E. Coli Whole Cell Model by Covert Lab / Time series run variant - Ciro Santilli">
<meta property="og:type" content="website">
<meta property="og:image" content="https://upload.wikimedia.org/wikipedia/commons/5/5f/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_acids_mass_fraction_summary.svg">
<meta property="og:url" content="https://cirosantilli.com/e-coli-whole-cell-model-by-covert-lab/time-series-run-variant">
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.10.1/css/all.min.css" integrity="sha512-9my9Mb2+0YO+I4PUCSwUYO7sEK21Y0STBAiFEYoWtd2VzLEZZ4QARDrZ30hdM1GlioHJ8o8cWQiy8IAb1hy/Hg==" crossorigin="anonymous" referrerpolicy="no-referrer">
<link rel="canonical" href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab/time-series-run-variant">
<style>@import "../_obb/dist/ourbigbook.css";

</style>
<link rel="stylesheet" type="text/css" href="../_raw/main.css">
<link rel="shortcut icon" type="image/x-icon" href="https://raw.githubusercontent.com/cirosantilli/media/master/ID_photo_of_Ciro_Santilli_taken_in_2013_square_398.jpg">
</head>
<body>
<header>
<div class="brand-group">
<a href="../" class="brand"><img src="https://raw.githubusercontent.com/cirosantilli/media/master/ID_photo_of_Ciro_Santilli_taken_in_2013_right_eye_200_100.jpg" loading="lazy" alt="ID photo of Ciro Santilli taken in 2013 right eye">Ciro Santilli</a>
<a href="https://ourbigbook.com/cirosantilli"><img src="https://raw.githubusercontent.com/cirosantilli/media/master/ourbigbook-logo-v1.svg" loading="lazy" alt="OurBigBook logo">OurBigBook.com</a>
<a class="font-awesome-container" href="https://stackoverflow.com/users/895245"><i class="fab fa-stack-overflow fa-fw icon"></i></a>
<a class="font-awesome-container" href="https://github.com/cirosantilli"><i class="fab fa-github fa-fw icon"></i></a>
<a class="font-awesome-container" href="https://www.linkedin.com/in/cirosantilli"><i class="fab fa-linkedin fa-fw icon"></i></a>
<a class="font-awesome-container" href="https://www.youtube.com/c/CiroSantilli"><i class="fab fa-youtube fa-fw icon"></i></a>
<a class="font-awesome-container" href="https://twitter.com/cirosantilli"><i class="fab fa-twitter fa-fw icon"></i></a>
<a class="font-awesome-container" href="https://www.zhihu.com/people/cirosantilli/activities"><i class="fab fa-zhihu fa-fw icon"></i></a>
<a class="font-awesome-container" href="https://www.weibo.com/p/1005055601627311"><i class="fab fa-weibo fa-fw icon"></i></a>
<a href="../sponsor"><span class="icon">$£</span>&nbsp;Sponsor</a>
<a href="https://github.com/cirosantilli/china-dictatorship"><span class="icon">中国</span>独裁统治&nbsp;China Dictatorship 新疆改造中心、六四事件、法轮功、郝海东、709大抓捕、2015巴拿马文件 邓家贵、低端人口、西藏骚乱</a>
</div>
</header>
<main class="ourbigbook">
<div class="h top" id="time-series-run-variant"><div class="notnav"><h1><a href="../e-coli-whole-cell-model-by-covert-lab">E. Coli Whole Cell Model by Covert Lab</a><span class="meta"> / </span><a href="">Time series run variant</a></h1></div><nav class="h-nav h-nav-toplevel"><div class="nav ancestors"><a  href="#_ancestors">&nbsp;...</a><a href="../taxonomy#bacteria"> Bacteria</a><a href="../taxonomy#list-of-bacteria"> List of bacteria</a><a href="../taxonomy#escherichia-coli"> Escherichia coli</a><a href="../taxonomy#e-coli-whole-cell-simulation"> E. Coli whole cell simulation</a><a href="../e-coli-whole-cell-model-by-covert-lab"> E. Coli Whole Cell Model by Covert Lab</a><a href="../e-coli-whole-cell-model-by-covert-lab#run-variants"> Run variants</a></div><div class="nav"><a href="https://ourbigbook.com/cirosantilli/e-coli-whole-cell-model-by-covert-lab/time-series-run-variant"><img src="../_obb/logo.svg" class="logo" alt=""> OurBigBook.com</a><a class="nosplit" href="../e-coli-whole-cell-model-by-covert-lab#time-series-run-variant"></a><span class="metrics"><span class="wcntr"> Words: 741</span></span></div></nav></div><div class="p" id="_74">To modify the nutrients as a function of time, with To select a time series we can use something like:<div class="code" id="_75"><div><pre><code>python runscripts/manual/runSim.py --variant nutrientTimeSeries 25 25</code></pre></div></div>As mentioned in <code>python runscripts/manual/runSim.py --help</code>, <code>nutrientTimeSeries</code> is one of the choices from <a href="https://github.com/CovertLab/WholeCellEcoliRelease/blob/7e4cc9e57de76752df0f4e32eca95fb653ea64e4/models/ecoli/sim/variants/__init__.py#L57">github.com/CovertLab/WholeCellEcoliRelease/blob/7e4cc9e57de76752df0f4e32eca95fb653ea64e4/models/ecoli/sim/variants/__init__.py#L57</a></div><div class="p" id="_76"><code>25 25</code> means to start from index 25 and also end at 25, so running just one simulation. <code>25 27</code> would run 25 then 26 and then 27 for example.</div><div class="p" id="_77">The timeseries with index 25 is <code>reconstruction/ecoli/flat/condition/timeseries/000025_cut_aa.tsv</code> and contains<div class="code" id="_78"><div><pre><code>"time (units.s)" "nutrients"
0 "minimal_plus_amino_acids"
1200 "minimal"</code></pre></div></div>so we understand that it starts with extra <a href="../protein#amino-acid">amino acids</a> in the medium, which benefit the cell, and half way through those are removed at time 1200s = 20 minutes. We would therefore expect the cell to start expressing amino acid production genes exactly at that point.</div><div class="p" id="_79"><code>nutrients</code> likely means <code>condition</code> in that file however, see bug report with <code>1 1</code> failing:  <a href="https://github.com/CovertLab/WholeCellEcoliRelease/issues/24">github.com/CovertLab/WholeCellEcoliRelease/issues/24</a></div><div class="p" id="_80">When we do this the simulation ends in:<div class="code" id="_81"><div><pre><code>Simulation finished:
 - Length: 0:34:23
 - Runtime: 0:08:03</code></pre></div></div>so we see that the doubling time was faster than the one with minimal conditions of <code>0:42:49</code>, which makes sense, since during the first 20 minutes the cell had extra <a href="../protein#amino-acid">amino acid</a> nutrients at its disposal.</div><div class="p" id="_82">The output directory now contains simulation output data under <code>out/manual/nutrientTimeSeries_000025/</code>. Let's run analysis and plots for that:<div class="code" id="_83"><div><pre><code>python runscripts/manual/analysisVariant.py &amp;&amp;
python runscripts/manual/analysisCohort.py --variant 25 &amp;&amp;
python runscripts/manual/analysisMultigen.py --variant 25 &amp;&amp;
python runscripts/manual/analysisSingle.py --variant 25</code></pre></div></div></div><div class="p" id="_84">We can now compare the outputs of this run to the default <code>wildtype_000000</code> run from <a href="../e-coli-whole-cell-model-by-covert-lab#install-and-first-run">Section "Install and first run"</a>.</div><div class="list"><ul id="_85"><li id="_86"><div class="p" id="_87"><code>out/manual/plotOut/svg_plots/massFractionSummary.svg</code>: because we now have two variants in the same <code>out/</code> folder, <code>wildtype_000000</code> and <code>nutrientTimeSeries_000025</code>, we now see a side by side comparision of both on the same graph!</div><div class="p" id="_88">The run variant where we started with amino acids initially grows faster as expected, because the cell didn't have to make it's own amino acids, so growth is a bit more efficient.</div><div class="p" id="_89">Then, at 20 minutes, which is about 0.3 hours, we see that the cell starts growing a bit less fast as the slope of the curve decreases a bit, because we removed that free amino acid supply.</div><div class="figure"><figure id="image-minimal-condition-vs-amino-acid-cut-mass-fraction-plot"><div class="float-wrap"><a href="https://upload.wikimedia.org/wikipedia/commons/5/5f/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_acids_mass_fraction_summary.svg"><img src="https://upload.wikimedia.org/wikipedia/commons/5/5f/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_acids_mass_fraction_summary.svg" loading="lazy" height="600"></a></div><figcaption><a href="#image-minimal-condition-vs-amino-acid-cut-mass-fraction-plot"><span class="caption-prefix">Figure 2. </span></a><div class="title">Minimal condition vs amino acid cut mass fraction plot</div>. <a href="https://commons.wikimedia.org/wiki/File:E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_acids_mass_fraction_summary.svg">Source</a>. From file <code>out/manual/plotOut/svg_plots/massFractionSummary.svg</code>.</figcaption></figure></div></li></ul></div><div class="p" id="_90">The following plots from under <code>out/manual/wildtype_000000/000000/{generation_000000,nutrientTimeSeries_000025}/000000/plotOut/svg_plots</code> have been manually joined side-by-side with:<div class="code" id="_91"><div><pre><code>for f in out/manual/wildtype_000000/000000/generation_000000/000000/plotOut/svg_plots/*; do
  echo $f
  svg_stack.py \
    --direction h \
    out/manual/wildtype_000000/000000/generation_000000/000000/plotOut/svg_plots/$(basename $f) \
    out/manual/nutrientTimeSeries_000025/000000/generation_000000/000000/plotOut/svg_plots/$(basename $f) \
    &gt; tmp/$(basename $f)
done</code></pre></div></div></div><div class="figure"><figure id="image-amino-acid-counts" class="multiline"><div class="float-wrap"><a href="https://upload.wikimedia.org/wikipedia/commons/e/e1/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_acids_amino_acid_counts.svg"><img src="https://upload.wikimedia.org/wikipedia/commons/e/e1/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_acids_amino_acid_counts.svg" loading="lazy" height="800"></a></div><figcaption><a href="#image-amino-acid-counts"><span class="caption-prefix">Figure 3. </span></a><div class="title">Amino acid counts</div>. <a href="https://commons.wikimedia.org/wiki/File:E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_acids_amino_acid_counts.svg">Source</a>. <code>aaCounts.svg</code>:<div class="list"><ul id="_92"><li id="_93">default: quantities just increase</li><li id="_94">amino acid cut: there is an abrupt fall at 20 minutes when we cut off external supply, presumably because it takes some time for the cell to start producing its own</li></ul></div></figcaption></figure></div><div class="p" id="_95"><div class="figure"><figure id="image-external-exchange-fluxes-of-amino-acids" class="multiline"><div class="float-wrap"><a href="https://upload.wikimedia.org/wikipedia/commons/7/7e/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_exchange_fluxes_of_amino_acids.svg"><img src="https://upload.wikimedia.org/wikipedia/commons/7/7e/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_exchange_fluxes_of_amino_acids.svg" loading="lazy" height="800"></a></div><figcaption><a href="#image-external-exchange-fluxes-of-amino-acids"><span class="caption-prefix">Figure 4. </span></a><div class="title">External exchange fluxes of amino acids</div>. <a href="https://commons.wikimedia.org/wiki/File:E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_exchange_fluxes_of_amino_acids.svg">Source</a>. <code>aaExchangeFluxes.svg</code>:<div class="list"><ul id="_96"><li id="_97">default: no exchanges</li><li id="_98"><div class="p" id="_99">amino acid cut: for all graphs except <a href="../protein#phenylalanine">phenylalanine</a> (PHE), either the cell was intaking the AA (negative flux), and that intake goes to 0 when the supply is cut, or the flux is always 0.</div><div class="p" id="_100">For PHE however, the flux is at all times, except shortly after the cut. Why? And why there was no excretion on the default conditions?</div></li></ul></div></figcaption></figure></div><div class="figure"><figure id="image-evaluation-time"><div class="float-wrap"><a href="https://upload.wikimedia.org/wikipedia/commons/d/d6/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_evaluation_time.svg"><img src="https://upload.wikimedia.org/wikipedia/commons/d/d6/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_evaluation_time.svg" loading="lazy" height="315"></a></div><figcaption><a href="#image-evaluation-time"><span class="caption-prefix">Figure 5. </span></a><div class="title">Evaluation time</div>. <a href="https://commons.wikimedia.org/wiki/File:E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_evaluation_time.svg">Source</a>. <code>evaluationTime.svg</code>: this has nothing to do with biology, but it is rather a <a href="../software#profiling-computer-programming">profile</a> of the program runtime. We can see that the simulation gets slower and slower as time passes, presumably because there are more and more molecules to simulate.</figcaption></figure></div></div><div class="figure"><figure id="image-mrna-count-of-highly-expressed-mrnas" class="multiline"><div class="float-wrap"><a href="https://upload.wikimedia.org/wikipedia/commons/7/7a/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_mrna_count_of_highly_expressed_mRNAs.svg"><img src="https://upload.wikimedia.org/wikipedia/commons/7/7a/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_mrna_count_of_highly_expressed_mRNAs.svg" loading="lazy" height="800"></a></div><figcaption><a href="#image-mrna-count-of-highly-expressed-mrnas"><span class="caption-prefix">Figure 6. </span></a><div class="title">mRNA count of highly expressed mRNAs</div>. <a href="https://commons.wikimedia.org/wiki/File:E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_mrna_count_of_highly_expressed_mRNAs.svg">Source</a>. From file <code>expression_rna_03_high.svg</code>. Each of the entries is a <a href="../dna#gene">gene</a> using the conventional gene naming convention of <code>xyzW</code>, e.g. here's the <a href="../biology#biocyc">BioCyc</a> for the first entry, <code>tufA</code>: <a href="https://biocyc.org/gene?orgid=ECOLI&amp;id=EG11036">biocyc.org/gene?orgid=ECOLI&amp;id=EG11036</a>, which comments <div><blockquote id="_101">Elongation factor Tu (EF-Tu) is the most abundant protein in E. coli.</blockquote></div> and <div><blockquote id="_102">In E. coli, EF-Tu is encoded by two genes, tufA and tufB</blockquote></div>. What they seem to mean is that tufA and tufB are two similar molecules, either of which can make up the <a href="../cell#ef-tu">EF-Tu</a> of the <a href="../taxonomy#escherichia-coli">E. Coli</a>, which is an important part of <a href="../cell#translation-biology">translation</a>.</figcaption></figure></div><div class="figure"><figure id="image-external-exchange-fluxes" class="multiline"><div class="float-wrap"><a href="https://upload.wikimedia.org/wikipedia/commons/4/49/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_exchange_fluxes.svg"><img src="https://upload.wikimedia.org/wikipedia/commons/4/49/E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_exchange_fluxes.svg" loading="lazy" height="1000"></a></div><figcaption><a href="#image-external-exchange-fluxes"><span class="caption-prefix">Figure 7. </span></a><div class="title">External exchange fluxes</div>. <a href="https://commons.wikimedia.org/wiki/File:E._Coli_Whole_Cell_model_by_Covert_Lab_minimal_nutrients_vs_cut_amino_external_exchange_fluxes.svg">Source</a>. <div class="p" id="_103"><code>mediaExcange.svg</code>: this one is similar to <code>aaExchangeFluxes.svg</code>, but it also tracks other substances. The color version makes it easier to squeeze more substances in a given space, but you lose the shape of curves a bit. The title seems reversed: red must be excretion, since that's where <a href="../chemistry#glucose">glucose</a> (GLC) is.</div><div class="p" id="_104">The substances are different between the default and amino acid cut graphs, they seem to be the most exchanged substances. On the amino cut graph, first we see the cell intaking most (except <a href="../protein#phenylalanine">phenylalanine</a>, which is excreted for some reason). When we cut amino acids, the uptake of course stops.</div></figcaption></figure></div><h2 id="_ancestors"><a href="#_ancestors"><span class="fa-solid-900 icon"></span> Ancestors <span class="meta">(12)</span></a></h2><div class="list"><ol><li><a href="../e-coli-whole-cell-model-by-covert-lab#run-variants">Run variants</a></li><li><a href="../e-coli-whole-cell-model-by-covert-lab">E. Coli Whole Cell Model by Covert Lab</a></li><li><a href="../taxonomy#e-coli-whole-cell-simulation">E. Coli whole cell simulation</a></li><li><a href="../taxonomy#escherichia-coli">Escherichia coli</a></li><li><a href="../taxonomy#list-of-bacteria">List of bacteria</a></li><li><a href="../taxonomy#bacteria">Bacteria</a></li><li><a href="../taxonomy#species">Species</a></li><li><a href="../taxonomy">Taxonomy</a></li><li><a href="../biology">Biology</a></li><li><a href="../science#natural-science">Natural science</a></li><li><a href="../science">Science</a></li><li><a href=".."><span title="Home" class="fa-solid-900 icon"></span> Home</a></li></ol></div><h2 id="_incoming-links"><a href="#_incoming-links"><span title="Incoming links" class="fa-solid-900 icon"></span> Incoming links <span class="meta">(2)</span></a></h2><div class="list"><ul><li><a href="../e-coli-whole-cell-model-by-covert-lab#mass-fraction-summary-plot-analysis">Mass fraction summary plot analysis</a></li><li><a href="../e-coli-whole-cell-model-by-covert-lab#other-run-variants">Other run variants</a></li></ul></div>
</main>
<footer>
<div>Powered by <a href="https://docs.ourbigbook.com">OurBigBook</a></div>
<div>License: <a href="https://creativecommons.org/licenses/by-sa/4.0/">CC BY-SA 4.0</a> unless noted</div>
<div><a href="https://github.com/cirosantilli/cirosantilli.github.io/issues">Suggestions and corrections</a></div>
<div><a href="../contact">Contact Ciro Santilli</a></div>
<div><a href="../_dir">Website source code</a></div>
<div><a href="https://github.com/cirosantilli/cirosantilli.github.io">Website source code on GitHub</a></div>
<div><a href="../_file/e-coli-whole-cell-model-by-covert-lab.bigb">Source code for this page: e-coli-whole-cell-model-by-covert-lab.bigb</a></div>
<div><a href="https://github.com/cirosantilli/cirosantilli.github.io/blob/a4d833d32a963800b7b90b69207f6802c6bb72d2/e-coli-whole-cell-model-by-covert-lab.bigb">Source code for this page on GitHub</a></div>
<div>Cite with: <a href="https://zenodo.org/badge/latestdoi/16453261">this DOI</a></div>
<div><img src="https://raw.githubusercontent.com/cirosantilli/media/master/ID_photo_of_Ciro_Santilli_taken_in_2013_left_eye_200_100.jpg" loading="lazy" alt="ID photo of Ciro Santilli taken in 2013 right eye"></div>
</footer>
<script>
window.ourbigbook_split_headers = true;
window.ourbigbook_html_x_extension = false;
window.ourbigbook_redirect_prefix = "e-coli-whole-cell-model-by-covert-lab/";
</script>
<script src="../_obb/dist/ourbigbook_runtime.js"></script><script>ourbigbook_runtime.ourbigbook_runtime()</script><script src="../_raw/main.js"></script>
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
ga('create', 'UA-47867706-1', 'auto');
ga('send', 'pageview');
</script>
<script async src="https://www.googletagmanager.com/gtag/js?id=G-DEE2HEJW9X"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());
  gtag('config', 'G-DEE2HEJW9X');
</script>
<script src="https://giscus.app/client.js"
        data-repo="cirosantilli/cirosantilli.github.io"
        data-repo-id="MDEwOlJlcG9zaXRvcnkxNjQ1MzI2MQ=="
        data-category="giscus"
        data-category-id="DIC_kwDOAPsOjc4CZ6zZ"
        data-mapping="pathname"
        data-strict="0"
        data-reactions-enabled="1"
        data-emit-metadata="0"
        data-input-position="bottom"
        data-theme="dark_high_contrast"
        data-lang="en"
        crossorigin="anonymous"
        async>
</script>
</body>
</html>
